internal/pkg/gateway/commit/blockparser.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	"errors"
+
+	"github.com/golang/protobuf/proto"
+	commonProto "github.com/hyperledger/fabric-protos-go/common"
+	peerProto "github.com/hyperledger/fabric-protos-go/peer"
+	ghErrors "github.com/pkg/errors"
+)
+
+type blockParser struct {
+	Block *commonProto.Block
+}
+
+func (parser *blockParser) TransactionValidationCodes() (map[string]peerProto.TxValidationCode, error) {
+	results := make(map[string]peerProto.TxValidationCode)
+
+	envelopes := parser.Block.Data.Data
+	for i, envelopeBytes := range envelopes {
+		channelHeader, err := unmarshalChannelHeader(envelopeBytes)
+		if err != nil {
+			return nil, ghErrors.WithMessagef(err, "failed to unmarshal channel header from envelope at index %v in block number %v",
```

Comment:
In general, we use github.com/pkg/errors for our errors in Fabric. If you really want to use vanilla `errors` I suppose that's not the end of the world, but let's not use both.

Reply:
I'll change to never use Go's own error APIs. Can we add the need to use GitHub's error package to the project coding guidelines?